common/src/main/java/us/ilite/common/LoggedData.java

```diff
@@ -0,0 +1,148 @@
+
+    //Stores writers per codex needed for CSV logging
+    private Map<String, Writer> mWriters = new HashMap<String, Writer>();
+
+    private List<CodexNetworkTablesParser> mLoggedCodexes;
+
+    public LoggedData() {
+        //Add new codexes as we support more into this list
+        mLoggedCodexes = Arrays.asList(
+            new CodexNetworkTablesParser<EGyro>(imu),
+            new CodexNetworkTablesParser<EDriveData>(drive),
+            new CodexNetworkTablesParser<ELogitech310>(driverinput, "DRIVER"),
+            new CodexNetworkTablesParser<ELogitech310>(operatorinput, "OPERATOR")
+        );
+
+        //This loop makes a Writer for each parser and sticks it into mWriters
+        for (CodexNetworkTablesParser parser : mLoggedCodexes) {
+            try {
+                File file = parser.file();
+                handleCreation(file);
+                mWriters.put(parser.getCSVIdentifier(), new BufferedWriter(new FileWriter(parser.file())));
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Translate NT to on-computer codex for each CodexNetworkTablesParser in mLoggedCodexes
+     */
+    public void logFromNetworkTables() {
+        mLoggedCodexes.forEach(c -> c.parseFromNetworkTables());
+    }
+
+    /**
+     * Makes a csv file and writes the starting row/header for each CodexNetworkTablesParser in mLoggedCodexes
+     */
+    public void logFromCodexToCSVHeader() {
+        for (CodexNetworkTablesParser parser : mLoggedCodexes) {
+            try {
+                Writer logger = mWriters.get(parser.getCSVIdentifier());
+                logger.append(parser.codexToCSVHeader());
+                logger.flush();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    /**
+     * Logs codex values to its corresponding csv
+     */
+    public void logFromCodexToCSVLog() {
+        for (CodexNetworkTablesParser parser : mLoggedCodexes) {
+            try {
+                Writer logger = mWriters.get(parser.getCSVIdentifier());
+                logger.append(parser.codexToCSVLog());
+                logger.flush();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+    /**
+     * Closes all the writers in mWriters
+     */
+    public void closeWriters() {
+        for (Writer writer : mWriters.values()) {
+            try {
+                writer.close();
+            }
+            catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Makes the log file if it doesn't already exist
+     */
+    private void handleCreation(File pFile) throws IOException {
+        //Makes every folder before the file if the CSV's parent folder doesn't exist
+        if(!pFile.getParentFile().exists()) pFile.getParentFile().mkdirs();
+
```

Comment:
It'd be a little better just for readability to put if statements on separate lines as the calls inside the statements

Reply:
I got the comments explaining the *if statement* for this reason. Do you think I should still separate it into multiple lines?

Reply:
yes